test/pkg/firewall/firewall.go

```diff
@@ -0,0 +1,212 @@
+package firewall
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+
+	"github.com/openshift-kni/commatrix/pkg/types"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func ApplyRulesToNode(cs *client.ClientSet, ns string, NFTtable []byte, nodeName string, wg *sync.WaitGroup) error {
+	defer wg.Done()
+
+	utilsHelpers := utils.New(cs)
+
+	debugPod, err := utilsHelpers.CreatePodOnNode(nodeName, ns, consts.DefaultDebugPodImage)
+	if err != nil {
+		return fmt.Errorf("failed to create debug pod on node %s: %w", nodeName, err)
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	command := []string{
+		"bash", "-c",
+		fmt.Sprintf("mkdir -p nft && echo '%s' > nft/firewall.nft", string(NFTtable)),
+	}
+
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to save rule set to file on node %s: %w", nodeName, err)
+	}
+
+	if err := installNFT(debugPod, utilsHelpers); err != nil {
+		return err
+	}
+
+	command = []string{
+		"bash", "-c", "/usr/sbin/nft -f nft/firewall.nft",
+	}
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to apply rule set %q on node %s: %w", command, nodeName, err)
+	}
+
+	output, err := nftList(debugPod, utilsHelpers)
```

Comment:
this functions needs to be part of the firewall struct object so you only send the debugPod